test/Calc.mo

```diff
@@ -24,12 +34,26 @@ actor {
                            #named("d", #sub(#num(5), #named("e", #div(#num(4), #num(2)))))
                       ))));
 
+    Debug.print "(trivial: exp done.)";
+
     calc.engine.draw().logEventLast();
 
+    Debug.print "calc.engine.draw().logEventLast() done";
+
     // to do -- change the expression somehow
 
     let res = calc.eval(exp);
+
+    Debug.print "calc.eval(exp) done";
+
     calc.engine.draw().logEventLast();
-    calc.engine.draw().getResult()
+
+    Debug.print "calc.engine.draw().logEventLast() done";
+
+    let r = calc.engine.draw().getResult();
+
+    Debug.print "calc.engine.draw().getResult() done";
```

Comment:
@chenyan-dfinity The test on my laptop and on CI do the same thing: They get past this line, but no further.  The call never returns, and the replica says something strange about buffer/queue sizes.

Reply:
Can you try `dfx canister call --output raw`? I suspect it's the pretty printer that is too slow.

Reply:
Oh, I see
Right.

Reply:
Will try now....Thanks!

Reply:
Bingo -- That was it.

And now that you remind me, I recall having this same issue months back, probably for the same datatype.  : )

I'll make a sticky note this time : ). Thanks again for the help!

Reply:
You can file an issue at candid repo. You are the only one who has large output so far :)